client/app/bundles/course/learning-map/containers/Dashboard/index.jsx

```diff
@@ -0,0 +1,245 @@
+import { useState } from 'react';
+import { Card, CardContent, CircularProgress, Icon } from '@mui/material';
+import { green, orange, red } from '@mui/material/colors';
+import { connect } from 'react-redux';
+import {
+  removeParentNode,
+  resetSelection,
+  toggleSatisfiabilityType,
+} from 'course/learning-map/actions';
+import ReactTooltip from 'react-tooltip';
+import ConfirmationDialog from 'lib/components/ConfirmationDialog';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+import PropTypes from 'prop-types';
+import { elementTypes, satisfiabilityTypes } from '../../constants';
+import translations from '../../translations.intl';
+import {
+  nodeShape,
+  responseShape,
+  selectedElementShape,
+} from '../../propTypes';
+
+const styles = {
+  content: {
+    alignItems: 'center',
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+  },
+  circularProgress: {
+    marginLeft: 12,
+    position: 'relative',
+  },
+  icon: {
+    cursor: 'pointer',
+    fontSize: '18px',
+    marginLeft: '20px',
+    padding: '0px',
+  },
+  wrapper: {
+    bottom: 0,
+    position: 'sticky',
+    textAlign: 'center',
+    zIndex: 999,
+  },
+};
+
+const Dashboard = (props) => {
+  const { dispatch, intl, isLoading, nodes, response, selectedElement } = props;
+
+  const [deleteArrowConfirmation, setDeleteArrowConfirmation] = useState(false);
+  const isEmptyResponse =
+    response &&
+    Object.keys(response).length === 0 &&
+    Object.getPrototypeOf(response) === Object.prototype;
```

Comment:
Hmm this check feels a bit weird haha, but I guess if it works, that's fine.

Reply:
Basically to check whether the object is empty